cinder/volume/drivers/nexenta/ns5/nfs.py

```diff
@@ -330,7 +331,9 @@ def migrate_volume(self, ctxt, volume, host):
     def initialize_connection(self, volume, connector):
         LOG.debug('Initialize volume connection for %s', volume['name'])
         url = 'hpr/activate'
-        data = {'datasetName': volume['provider_location'].split(':/')[1]}
+        pool, fs = self.share.split('/', 1)
+        data = {'datasetName': '%s/%s/%s' % (
```

Comment:
Split with '/' and then join all items with '/'. 
Why not:
data = {'datasetName': '%s/%s' % self.share, volume['name']}
instead of lines 334-336?

Reply:
you're right, fixed